pkg/operator/staticresource.go

```diff
@@ -0,0 +1,171 @@
+	name string,
+	operatorNamespace string,
+	operatorClient operatorv1helpers.OperatorClientWithFinalizers,
+	kubeClient kubernetes.Interface,
+	informers operatorv1helpers.KubeInformersForNamespaces,
+	recorder events.Recorder,
+	csiDriver *storagev1.CSIDriver,
+	nodeServiceAccount *corev1.ServiceAccount,
+	nodeRole *rbacv1.ClusterRole,
+	nodeRoleBinding *rbacv1.ClusterRoleBinding,
+) factory.Controller {
+	c := &CSIStaticResourceController{
+		operatorName:       name,
+		operatorNamespace:  operatorNamespace,
+		operatorClient:     operatorClient,
+		kubeClient:         kubeClient,
+		eventRecorder:      recorder,
+		csiDriver:          csiDriver,
+		nodeServiceAccount: nodeServiceAccount,
+		nodeRole:           nodeRole,
+		nodeRoleBinding:    nodeRoleBinding,
+	}
+
+	operatorInformers := []factory.Informer{
+		operatorClient.Informer(),
+		informers.InformersFor(operatorNamespace).Core().V1().ServiceAccounts().Informer(),
+		informers.InformersFor(operatorNamespace).Storage().V1().CSIDrivers().Informer(),
+		informers.InformersFor(operatorNamespace).Rbac().V1().ClusterRoles().Informer(),
+		informers.InformersFor(operatorNamespace).Rbac().V1().ClusterRoleBindings().Informer(),
+	}
+	return factory.New().
+		WithSyncDegradedOnError(operatorClient).
+		WithInformers(operatorInformers...).
+		WithSync(c.sync).
+		ResyncEvery(time.Minute).
+		ToController(name, recorder.WithComponentSuffix("csi-static-resource-controller"))
+}
+
+func (c *CSIStaticResourceController) sync(ctx context.Context, controllerContext factory.SyncContext) error {
+	opSpec, opStatus, _, err := c.operatorClient.GetOperatorState()
+	if apierrors.IsNotFound(err) {
+		// TODO: proceed with removal?
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	if opSpec.ManagementState != opv1.Managed {
+		return nil
+	}
+
+	meta, err := c.operatorClient.GetObjectMeta()
+	if err != nil {
+		return err
+	}
+	if meta.DeletionTimestamp != nil {
+		return c.syncDeleting(ctx, opSpec, opStatus, controllerContext)
+	}
+	return c.syncManaged(ctx, opSpec, opStatus, controllerContext)
+}
+
+func (c *CSIStaticResourceController) syncManaged(ctx context.Context, opSpec *opv1.OperatorSpec, opStatus *opv1.OperatorStatus, controllerContext factory.SyncContext) error {
+	err := operatorv1helpers.EnsureFinalizer(c.operatorClient, c.operatorName)
+	if err != nil {
+		return err
+	}
+
+	var errs []error
+	_, _, err = resourceapply.ApplyCSIDriver(ctx, c.kubeClient.StorageV1(), c.eventRecorder, c.csiDriver)
+	if err != nil {
+		errs = append(errs, err)
+	}
+	_, _, err = resourceapply.ApplyClusterRole(ctx, c.kubeClient.RbacV1(), c.eventRecorder, c.nodeRole)
+	if err != nil {
+		errs = append(errs, err)
+	}
+	_, _, err = resourceapply.ApplyClusterRoleBinding(ctx, c.kubeClient.RbacV1(), c.eventRecorder, c.nodeRoleBinding)
+	if err != nil {
+		errs = append(errs, err)
+	}
+	_, _, err = resourceapply.ApplyServiceAccount(ctx, c.kubeClient.CoreV1(), c.eventRecorder, c.nodeServiceAccount)
```

Comment:
Is it possible to use `ApplyDirectly` instead? Anyway, we can do it in a separate PR later on.

Reply:
There is no `DeleteDirectly` for `syncDeleting`. I would need to either add it to library-go or parse the yaml files to get name of the objects there.